diff -uprN psm-3.3_raw/psm_diags.c psm-3.3/psm_diags.c
--- psm-3.3_raw/psm_diags.c	2014-08-21 04:00:18.000000000 +0800
+++ psm-3.3/psm_diags.c	2020-03-13 21:35:28.340564067 +0800
@@ -280,10 +280,11 @@ memcpy_check_size (memcpy_fn_t fn, int *
   if (USE_MALLOC) {
     src = psmi_malloc(PSMI_EP_NONE, UNDEFINED, size);
     dst = psmi_malloc(PSMI_EP_NONE, UNDEFINED, size);
-    if (src == NULL || dst == NULL) 
+    if (src == NULL || dst == NULL) {
       if (src) psmi_free(src);
       if (dst) psmi_free(dst);
       return -1;
+    }
   }
   else {
     void *src_p, *dst_p;
